specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/iotSecuritySolutionAnalytics.json

```diff
@@ -0,0 +1,916 @@
+        "topAlertedDevices": {
+          "description": "The list of top 3 devices with the most attacked.",
+          "type": "object",
+          "$ref": "#/definitions/IoTSecurityAlertedDevicesList"
+        },
+        "mostPrevalentDeviceAlerts": {
+          "description": "The list of most prevalent 3 alerts.",
+          "type": "object",
+          "$ref": "#/definitions/IoTSecurityDeviceAlertsList"
+        },
+        "mostPrevalentDeviceRecommendations": {
+          "description": "The list of most prevalent 3 recommendations.",
+          "type": "object",
+          "$ref": "#/definitions/IoTSecurityDeviceRecommendationsList"
+        }
+      }
+    },
+    "IoTSecuritySolutionAnalyticsModelList": {
+      "description": "List of Security Analytics of a security solution",
+      "required": [
+        "value"
+      ],
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "List of Security Analytics of a security solution",
+          "items": {
+            "$ref": "#/definitions/IoTSecuritySolutionAnalyticsModel"
+          }
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The URI to fetch the next page."
+        }
+      }
+    },
+    "IoTSecurityAggregatedAlertList": {
+      "description": "List of IoT aggregated security alerts",
+      "required": [
+        "value"
+      ],
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "List of aggregated alerts data",
+          "items": {
+            "$ref": "#/definitions/IoTSecurityAggregatedAlert"
+          }
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The URI to fetch the next page."
+        }
+      }
+    },
+    "IoTSecurityAggregatedRecommendationList": {
+      "description": "List of IoT aggregated security recommendations",
+      "required": [
+        "value"
+      ],
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "List of aggregated alerts data",
+          "items": {
+            "$ref": "#/definitions/IoTSecurityAggregatedRecommendation"
+          }
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The URI to fetch the next page."
+        }
+      }
+    },
+    "IoTSecurityAlertedDevicesList": {
+      "description": "List of devices with the count of raised alerts",
+      "required": [
+        "value"
+      ],
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "List of aggregated alerts data",
+          "items": {
+            "$ref": "#/definitions/IoTSecurityAlertedDevice"
+          }
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The URI to fetch the next page."
+        }
```

Comment:
This pattern (`value` array and `nextLink` property) is intended for collection GET calls. As far as I can tell this is part of a sub-property of a resource and is not needed. You should simplify this by changing the `topAlertedDevices` to an array of `/definitions/IoTSecurityAlertedDevice`.

Reply:
Currently we only gives the top 3 devices alerts but n the future we want to support queries on alerted devices and this can be a long list. so I have added support for paging at this object

Reply:
How are you planning to pass the nextLink to your service on a GET call? The standard pattern only works when paging through the resource itself. It doesn't work with paging through sub-properties of an individual resource.

---
In reply to: [308873954](https://github.com/Azure/azure-rest-api-specs/pull/6766#discussion_r308873954) [](ancestors = 308873954)

Reply:
I have removed it.